pareto/operational_water_management/operational_produced_water_optimization_model.py

```diff
@@ -2267,221 +1943,54 @@ def BidirectionalFlowRule2(model, l, l_tilde, t):
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_N and l_tilde in model.s_O:
-            if model.p_NOA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_F and l_tilde in model.s_CP:
-            if model.p_FCA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_R and l_tilde in model.s_N:
-            if model.p_RNA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_R and l_tilde in model.s_K:
-            if model.p_RKA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_N:
-            if model.p_SNA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_CP:
-            if model.p_SCA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_K:
-            if model.p_SKA[l, l_tilde]:
-                return (
-                    model.v_F_Capacity[l, l_tilde]
-                    == model.p_sigma_Pipeline[l, l_tilde]
-                    + model.v_S_PipelineCapacity[l, l_tilde]
-                )
-            else:
-                return Constraint.Skip
-        elif l in model.s_S and l_tilde in model.s_R:
-            if model.p_SRA[l, l_tilde]:
+    )
+
+    def PipelineCapacityExpansionRule(model, l, l_tilde):
```

Comment:
@nareshsusarla I went ahead and applied changes to this constraint in the operational model. Can you double check this looks as we discussed? Feel free to steal it for the strategic model as well.

Reply:
This is being applied to the strategic model in #248.